Keen/KeenHttpClient.cs

```diff
@@ -0,0 +1,205 @@
+            // like here?: https://msdn.microsoft.com/en-us/library/system.net.http.httpclientextensions.putasjsonasync(v=vs.118).aspx
+            using (var contentStream =
+                new StreamContent(new MemoryStream(Encoding.UTF8.GetBytes(content))))
+            {
+                // TODO : Amake sure this is the same as Add("content-type", "application/json")
+                contentStream.Headers.ContentType =
+                    new MediaTypeHeaderValue(KeenHttpClient.JSON_CONTENT_TYPE);
+
+                HttpRequestMessage post = KeenHttpClient.NewPost(resource, authKey);
+                post.Content = contentStream;
+
+                return await _httpClient.SendAsync(post).ConfigureAwait(false);
+
+                // TODO : Should we do the KeenUtil.CheckApiErrorCode() here?
+                // TODO : Should we check the if (!responseMsg.IsSuccessStatusCode) here too?
+                // TODO : If we centralize error checking in this class we could have variations
+                //     of these helpers that return string or JToken or JArray or JObject. It might
+                //     also be nice for those options to optionally hand back the raw
+                //     HttpResponseMessage in an out param if desired?
+                // TODO : Use CallerMemberNameAttribute to print error messages?
+                //     http://stackoverflow.com/questions/3095696/how-do-i-get-the-calling-method-name-and-type-using-reflection?noredirect=1&lq=1
+            }
+        }
+
+        /// <summary>
+        /// Create and send a DELETE request to the given relative resource using the given key for
+        /// authentication.
+        /// </summary>
+        /// <param name="resource">The relative resource to DELETE. Must be properly formatted as a
+        ///     relative Uri.</param>
+        /// <param name="authKey">The key to use for authenticating this request.</param>
+        /// <returns>The response message.</returns>
+        public Task<HttpResponseMessage> DeleteAsync(string resource, string authKey)
+        {
+            var url = new Uri(resource, UriKind.Relative);
+
+            return DeleteAsync(url, authKey);
+        }
+
+        /// <summary>
+        /// Create and send a DELETE request to the given relative resource using the given key for
+        /// authentication.
+        /// </summary>
+        /// <param name="resource">The relative resource to DELETE.</param>
+        /// <param name="authKey">The key to use for authenticating this request.</param>
+        /// <returns>The response message.</returns>
+        public Task<HttpResponseMessage> DeleteAsync(Uri resource, string authKey)
+        {
+            KeenHttpClient.RequireAuthKey(authKey);
+
+            HttpRequestMessage delete = KeenHttpClient.NewDelete(resource, authKey);
+
+            return _httpClient.SendAsync(delete);
+        }
+
+        private static HttpRequestMessage NewGet(Uri resource, string authKey)
```

Comment:
I originally thought there would be more involved in those methods, but it turns out not to be the case, so I'll simplify.